### DISCUSSION TITLE
Allow a migration to run multiple times

### DISCUSSION THREAD

**Discussion by jjwilliams42** [fluentmigrator/fluentmigrator#1409] [Q&A] [answered]
**Describe the question**
In the past, a typical pattern I have is to always run a set of migrations labeled as "Seed" data. Maybe this is the wrong term, but it refers to the more static reference data in my system. The database must always match my backing class.

In DbUp, there was a simple solution: Just don't write it to the Journal table:

```
DeployChanges.To.SqlDatabase(_cxn)
    .WithScriptsAndCodeEmbeddedInAssembly(typeof(Migrator).Assembly, x => x.Contains("-Seed-") || x.Contains("_Seed_"))
    .LogToConsole()
    .JournalTo(new NullJournal()) // We want seeds to always run (don't journal to the database)
    .LogTo(_dbUpLogger)
    .WithVariablesDisabled()
    .Build();
```

Is there an equivalent way to set a Null IVersionTableMetadata for FluentMigrator? Essentially saying: Don't write this migration to the Journal table.

**Documentation Pages You've Read So Far**
https://fluentmigrator.github.io/articles/version-table-metadata.html

Other things I've tried: Creating a NullVersionTableMetadata class, not setting anything on it. This did not work.

**Expected benefit**
If I got my answer to my question, I could continuously run a migration or set of migrations to keep certain static data always updated.

**Information (please complete the following information):**
 - OS: [ Windows 10 ]
 - Platform [.NET 5]
 - FluentMigrator version [ e.g 3.2.11 ]
 - FluentMigrator runner [ "in-process runner" ]
 - Database Management System [ SQLServer ]
 - Database Management System Version [ e.g., "SQL Server 2019 Edition" ]

**Comment by fubar-coder** [comment, marked as answer]
What about using a ["maintenance migration"](https://fluentmigrator.github.io/articles/glossary.html#maintenance)? Those are not added as migrations to the version table and are also configurable to be executed after all other DB migrations. They are (IIRC) sorted by name, so you can also name them S01Name, S02Name, etc... and they'll always be executed n this order.

**Comment by jzabroski** [comment]
I think you're asking a bit of an xy problem. There's no explanation in your question of why you don't want to journal your seed data. (Calling it SeedData or DataSeeding are both common.)

That said, maintenance Migrations are what you want. I would suggest taking advantage of dependency injection and injecting all instances of ISeedData into your test database. That way you keep the number of Maintenance classes low and you can have fine grained control of the sequence in which you insert data by having an Order property on ISeedData. It's also huge mentally to have a lot of maintenance Migrations. 

For non test environments, I just seed my tables in a separate, normal migration, usually one version number higher than my table creating script. I use Tags to avoid running normal seedlings in Test environments like UnitTest.

**Reply by jjwilliams42** [reply]
Yeah, I stated it a little backwards. Asking for a specific solution I guess you could say. 

The reason I would not want to journal the migrations is because I want them to always run. If a migration gets journaled to the database, it will never run again.

But maintenance migrations sounds like exactly what I need. Thanks!
